src/pages/index.js

```diff
@@ -60,33 +65,125 @@ const editModalDescriptionInput = editModal.querySelector(
 const cardForm = cardModal.querySelector(".modal__form");
 const cardSubmitButton = cardModal.querySelector(".modal__submit-button");
 const cardModalCloseButton = cardModal.querySelector(".modal__close-button");
 const cardNameInput = cardModal.querySelector("#add-card-name-input");
 const cardLinkInput = cardModal.querySelector("#add-card-link-input");
 
+// Avatar form elements
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarSubmitButton = avatarModal.querySelector(".modal__submit-button");
+const avatarModalCloseButton = avatarModal.querySelector(
+  ".modal__close-button"
+);
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+
+// Delete form elements
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form");
+
+const deleteModalCancelButton = deleteModal.querySelector(
+  ".modal__submit-button:last-of-type"
+);
+deleteModalCancelButton.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+const deleteModalCloseButton = deleteModal.querySelector(
+  ".modal__close-button"
+);
+deleteModalCloseButton.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+// Preview image popup elements
 const previewModal = document.querySelector("#preview-modal");
 const previewModalImageEl = previewModal.querySelector(".modal__image");
 const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
 const previewModalCloseButton = previewModal.querySelector(
   ".modal__close-button_preview"
 );
 
+//Card related elements
 const cardTemplate = document.querySelector("#card-template");
 const cardsList = document.querySelector(".cards__list");
 
+let selectedCard, selectedCardId;
+
+// Event Handlers
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
+
   const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardEl = getCardElement(inputValues);
-  cardsList.prepend(cardEl);
-  evt.target.reset();
-  disableButton(cardSubmitButton, settings);
-  closeModal(cardModal);
+
+  const cardSubmitButton = evt.submitter;
+  cardSubmitButton.textContent = "Saving...";
+
+  api
+    .addCard(inputValues)
+    .then((savedCard) => {
+      const cardEl = getCardElement(savedCard);
+      cardsList.prepend(cardEl);
+      evt.target.reset();
+
+      if (typeof disableButton === "function") {
+        disableButton(cardSubmitButton, settings);
+      }
+
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      cardSubmitButton.textContent = "Save";
+    });
 }
 
+function handleAvatarFormSubmit(evt) {
+  evt.preventDefault();
+
+  const avatarsubmitButton = evt.submitter;
+  avatarSubmitButton.textContent = "Saving...";
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      document.querySelector(".profile__avatar").src = data.avatar;
```

Comment:
`document.querySelector(".profile__avatar")`

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine